File: src/spec/resources/testprojects/nodetest/models/pet-food.ts
import { Expose, Type } from 'class-transformer';

/**
 * Food for pets, discriminated by foodType
 */
export type PetFoodType = DryFood | WetFood;

export class PetFood {
  static readonly ONE_OF_SCHEMAS: string[] = ['DryFood', 'WetFood'];
  static readonly DISCRIMINATOR_PROPERTY = 'foodType';
  static readonly DISCRIMINATOR_MAPPING: Record<string, string> = {
    dry: 'DryFood',
    wet: 'WetFood'
  };

  private actualInstance: PetFoodType;

  constructor(instance: PetFoodType) {
    this.actualInstance = instance;
  }

  getActualInstance(): PetFoodType {
    return this.actualInstance;
  }
}

File: src/spec/resources/testprojects/csharptest/src/PetstoreClient/Models/PetFood.cs
#pragma warning disable CA1002 // Do not expose generic lists
#pragma warning disable CA1711 // Identifiers should not have incorrect suffix
#pragma warning disable CA2227 // Collection properties should be read only

using System.Text.Json.Serialization;

namespace PetstoreClient.Models;

/// <summary>
/// Food for pets, discriminated by foodType
/// </summary>
[JsonPolymorphic(TypeDiscriminatorPropertyName = "foodType")]
[JsonDerivedType(typeof(DryFood), "dry")]
[JsonDerivedType(typeof(WetFood), "wet")]
public abstract class PetFood { }
